### PR TITLE
Awareness fix

### PR DESCRIPTION
**Pull request by nairrrahul**
fixed missing awareness handler in `index.ts`

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -311,8 +351,18 @@ export const jupyterCommentingPlugin: JupyterFrontEndPlugin<ICommentPanel> = {
       }
     });
 
+    let currAwareness: Awareness | null = null;
```
Comment (srirag-vuppala) [inline src/index.ts:354]: Why the null=null ?

File: src/index.ts [hunk 2]
```diff
@@ -333,7 +383,7 @@ export const jupyterCommentingPlugin: JupyterFrontEndPlugin<ICommentPanel> = {
       }
       editorWidget.editor.focus();
 
-      editorWidget.node.oncontextmenu = () => {
+      /*editorWidget.node.oncontextmenu = () => {
```
Comment (srirag-vuppala) [inline src/index.ts:386]: Maybe get rid of the comments if we don't require it for later?

File: src/widget.tsx [hunk 3]
```diff
@@ -964,6 +963,7 @@ export class CommentFileWidget extends Panel {
     const widget = factory.createWidget(comment, this.model);
 
     if (widget != null) {
+      console.log('inserting here!');
```
Comment (srirag-vuppala) [inline src/widget.tsx:966]: Might want to remove the console logs as well

File: package.json [hunk 4]
```diff
@@ -47,6 +47,8 @@
     "watch:labextension": "jupyter labextension watch ."
   },
   "dependencies": {
+    "@jupyterlab/rendermime": "^3.1.0-rc.1",
```
Comment (cameron-toy) [inline package.json:50]: This should be ^3.1.3

File: src/factory.ts [hunk 5]
```diff
@@ -325,12 +330,26 @@ export class TextSelectionCommentFactory extends ACommentFactory<CodeEditorWrapp
...
 
-    wrapper!.editor.model.selections.set(this._path, selections);
+    selectionsMap.set(
+      (wrapper.parent as DocumentWidget)?.context.path,
+      selections
+    );
 
     widget.disposed.connect(() => {
-      const sels = wrapper!.editor.model.selections.get(this._path) ?? [];
+      console.warn('disposing');
```
Comment (cameron-toy) [inline src/factory.ts:339]: remove console.warn

File: src/index.ts [hunk 6]
```diff
@@ -303,6 +303,46 @@ export const jupyterCommentingPlugin: JupyterFrontEndPlugin<ICommentPanel> = {
...
+          if (editorWidget == null) {
+            return;
+          }
+
+          const model = panel.model;
+          if (model == null) {
+            return;
+          }
+
+          const comments = model.comments;
```
Comment (cameron-toy) [inline src/index.ts:325]: I think this is unnecessary--it's for placing the comment at the right spot in the notebook files but doesn't do anything here

File: src/index.ts (hunk 1 above)
Comment (cameron-toy) [reply src/index.ts:354]: The type of `currAwareness` is `Awareness | null`. The initial value is null.

File: src/index.ts [hunk 7]
```diff
@@ -342,9 +392,41 @@ export const jupyterCommentingPlugin: JupyterFrontEndPlugin<ICommentPanel> = {
...
+      if (currAwareness != null) {
+        currAwareness.off('change', handler);
+      }
+
+      currAwareness = (editorWidget.editor.model.sharedModel as YFile)
+        .awareness;
+      currAwareness.on('change', handler);
     });
 
+    commentTracker.widgetAdded.connect((_, widget) =>
```
Comment (cameron-toy) [inline src/index.ts:426]: This just adds console noise and isn't needed

**Pull request merged**